### PR TITLE
Preliminary support for type aliases

### PR DESCRIPTION
**Pull request by ranjitjhala**
See 

* [pos](liquid-rust-tests/tests/pos/surface/alias00.rs)
* [neg](liquid-rust-tests/tests/neg/surface/alias00.rs)

Now also works across files, e.g.

* [library](liquid-rust-tests/tests/lib/nat.rs) 
* [client](liquid-rust-tests/tests/neg/surface/alias01.rs)

### CODE REVIEW

File: liquid-rust-syntax/src/lexer.rs [hunk 1]
```diff
@@ -11,6 +11,7 @@ use rustc_span::{symbol::kw, BytePos, Symbol};
 pub enum Token {
     Caret,
     EqEq,
+    Eq,
```
Comment (nilehmann) [inline liquid-rust-syntax/src/lexer.rs:14]: I think the problem is that you have to add a case to `Cursor::map_token` to add new tokens.

File: liquid-rust-syntax/src/lexer.rs [hunk 2]
```diff
@@ -27,6 +28,7 @@ pub enum Token {
     Ge,
     At,
     Fn,
+    Type,
```
Comment (nilehmann) [inline liquid-rust-syntax/src/lexer.rs:31]: This is probably a failed experiment?

File: liquid-rust-syntax/src/resolve.rs [hunk 3]
```diff
@@ -106,6 +106,7 @@ impl<'tcx> Resolver<'tcx> {
             }
             surface::Arg::StrgRef(loc, ty) => Ok(surface::Arg::StrgRef(loc, self.resolve_ty(ty)?)),
             surface::Arg::Ty(ty) => Ok(surface::Arg::Ty(self.resolve_ty(ty)?)),
+            _ => panic!("unexpected: resolve_arg"),
```
Comment (nilehmann) [inline liquid-rust-syntax/src/resolve.rs:109]: I prefer to be explicit about the missing case to catch errors if we add more variants.

**Review by nilehmann — APPROVED** [review]
Looks reasonable

File: liquid-rust-syntax/src/lexer.rs (hunk 1 above)
Comment (ranjitjhala) [reply liquid-rust-syntax/src/lexer.rs:14]: Ah yes, that fixes it!

File: liquid-rust-syntax/src/lexer.rs (hunk 2 above)
Comment (ranjitjhala) [reply liquid-rust-syntax/src/lexer.rs:31]: oops deleted!

File: liquid-rust-syntax/src/resolve.rs (hunk 3 above)
Comment (ranjitjhala) [reply liquid-rust-syntax/src/resolve.rs:109]: Good point, have fixed -- btw, note that if we had a refinements, we should be able to prove this arm is unreachable...

**Review by nilehmann — APPROVED** [review]
I was worried about name clashes between flux aliases and rust types, but I think annotating rust aliases fixes this issue. I like it!

**Pull request merged**